subprojects/kotlin-dsl-provider-plugins/src/test/kotlin/org/gradle/kotlin/dsl/provider/plugins/precompiled/PrecompiledScriptPluginTest.kt

```diff
@@ -114,7 +114,7 @@ class PrecompiledScriptPluginTest : TestWithTempFiles() {
                  *
                  * @see My_script_gradle
                  */
-                class MyScriptPlugin
+                public class MyScriptPlugin
```

Comment:
Nit: having also a separate test that checks specifically for the `public` modifier might be desirable, so that the tests have a more clearly expressed expectation of the modifier presence.

Reply:
An ideal test would be to use `explicitApi()` and try to compile, because that is exactly the issue. The rules of `explicitApi()` might change over time and we would want the test to fail if our generated code diverges from its rules due to an update.

Reply:
Yes, and I guess that's what @ephemient tried to achieve in `PrecompiledScriptPluginIntegrationTest`.